sdk/wallet-adapter/adapters/base-adapter/src/index.ts

```diff
@@ -27,18 +32,17 @@ export interface WalletAdapter {
     event: E,
     callback: WalletAdapterEvents[E]
   ) => () => void;
-  signTransaction(transaction: SignableTransaction): Promise<SignedTransaction>;
+  signTransaction(
+    transactionInput: WithOptionalChain<SuiSignTransactionInput>
+  ): Promise<SignedTransaction>;
   /**
    * Suggest a transaction for the user to sign. Supports all valid transaction types.
    */
   signAndExecuteTransaction(
-    transaction: SignableTransaction,
-    options?: {
-      requestType?: ExecuteTransactionRequestType;
-    }
+    transactionInput: WithOptionalChain<SuiSignAndExecuteTransactionInput>
   ): Promise<SuiTransactionResponse>;
 
-  getAccounts: () => Promise<SuiAddress[]>;
+  getAccounts: () => Promise<ReadonlyWalletAccount[]>;
```

Comment:
This should use the `WalletAccount` interface not the `ReadonlyWalletAccount` concrete implementation.